### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Extensive): Relax typeclass requirements

### PR DESCRIPTION
**Pull request by erdOne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Extensive.lean [hunk 1]
```diff
@@ -61,26 +61,42 @@ section Extensive
 
 variable {X Y : C}
 
+/-- A category has pullback of inclusions if it has all pullbacks along coproduct injections. -/
+class HasPullbackOfInclusions (C : Type u) [Category.{v} C] [HasBinaryCoproducts C] : Prop where
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Extensive.lean:65]:
```suggestion
class HasPullbacksOfInclusions (C : Type u) [Category.{v} C] [HasBinaryCoproducts C] : Prop where
```

File: Mathlib/CategoryTheory/Extensive.lean [hunk 2]
```diff
@@ -61,26 +61,42 @@ section Extensive
...
+  [hasPullbackInl : ∀ {X Y Z : C} (f : Z ⟶ X ⨿ Y), HasPullback coprod.inl f]
+
+attribute [instance] HasPullbackOfInclusions.hasPullbackInl
+
+/--
+A functor preserves pullback of inclusions if it preserves all pullbacks along coproduct injections.
+-/
+class PreservesPullbackOfInclusions {C : Type*} [Category C] {D : Type*} [Category D]
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Extensive.lean:73]:
```suggestion
class PreservesPullbacksOfInclusions {C : Type*} [Category C] {D : Type*} [Category D]
```

and similarly below

File: Mathlib/CategoryTheory/Extensive.lean [hunk 3]
```diff
@@ -387,12 +444,34 @@ instance finitaryExtensive_functor [HasPullbacks C] [FinitaryExtensive C] :
...
+  constructor
+  intros X Y c hc
+  refine IsVanKampenColimit.of_iso ?_ (hc.uniqueUpToIso (coprodIsCoprod X Y)).symm
+  have (i : Discrete WalkingPair) (Z : C) (f : Z ⟶ X ⨿ Y) :
+    PreservesLimit (cospan f ((BinaryCofan.mk coprod.inl coprod.inr).ι.app i)) F := by
+    rcases i with ⟨_|_⟩ <;> dsimp <;> infer_instance
+  refine (FinitaryExtensive.vanKampen _
+    (isColimitOfPreserves F (coprodIsCoprod X Y))).of_mapCocone F
+
+  -- ⟨fun c hc => (FinitaryExtensive.vanKampen _ (isColimitOfPreserves F hc)).of_mapCocone F⟩
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Extensive.lean:474]:
```suggestion
```

File: Mathlib/CategoryTheory/Extensive.lean (hunk 1 above)
Comment (dagurtomas) [reply Mathlib/CategoryTheory/Extensive.lean:65]: Because it has *all* pullbacks of *all* inclusions

**Comment by dagurtomas** [comment]
Otherwise, LGTM

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**